controller-client/src/lib.rs

```diff
@@ -604,6 +612,57 @@ impl ControllerClientImpl {
         })
     }
 
+    async fn call_list_streams(&self, scope: &Scope, token: &str) -> Result<Option<(Vec<String>, String)>> {
+        let operation_name = "ListStreams";
+        let request: StreamsInScopeRequest = StreamsInScopeRequest {
+            scope: Some(ScopeInfo::from(scope)),
+            continuation_token: Some(ContinuationToken {
+                token: token.to_string(),
+            }),
+        };
+        debug!(
+            "Triggering a request to the controller to list streams for scope {}",
+            scope
+        );
+
+        let op_status: StdResult<tonic::Response<StreamsInScopeResponse>, tonic::Status> =
+            self.get_controller_client().list_streams_in_scope(request).await;
+        match op_status {
+            Ok(streams_with_token) => {
+                let result = streams_with_token.into_inner();
+                let t: Vec<StreamInfo> = result.streams;
+                if t.is_empty() {
+                    // Empty result from the controller implies no further streams present.
+                    Ok(None)
+                } else {
+                    // update state with the new set of streams.
+                    let stream_list: Vec<String> =
+                        t.iter().map(|i: &StreamInfo| i.stream.to_string()).collect();
+
+                    let token: Option<ContinuationToken> = result.continuation_token;
+                    let ct: String = match token.map(|t| t.token) {
+                        None => {
+                            warn!(
+                                "None returned for continuation token list streams API for scope {}",
+                                scope
+                            );
+                            String::from("")
+                        }
```

Comment:
Doesn't this just mean there are no more items? Why the warning? and why the empty string?

Reply:
improved the code behaviour here.